libraries/rush-lib/src/logic/operations/IOperationExecutionResult.ts

```diff
@@ -40,6 +40,10 @@ export interface IOperationExecutionResult {
    * Object used to report a summary at the end of the Rush invocation.
    */
   readonly stdioSummarizer: StdioSummarizer;
+  /**
+   * Object used to collect problems (errors/warnings/info) encountered during the operation.
+   */
+  readonly problemCollector: ProblemCollector;
```

Comment:
Define an `IProblemCollector` interface with only the `getProblems()` method (or expose `getProblems()` as a method that uses the `problemCollector` as an implementation detail. I prefer not to expose the guts of that this is an entity that talks to the terminal on the public API contract.